#158487973 Invite online user as friend

ozimos opened the pull request:
#### What does this PR do?
This pull request adds the ability to search for online users and add them as friends
#### Description of Task to be completed?
- Adds friends to the user collection
- Adds /api/friends endpoint to allow adding a user as friends and getting a user's friends
- Adds notifications to the navbar when a friend is added
- Adds a modal to search for users
#### How should this be manually tested?
run npm test
#### Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
#158487973
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/27624208/42872306-7d87ea2c-8a74-11e8-9c45-3674a1471da5.png)
![image](https://user-images.githubusercontent.com/27624208/42872324-8b1a3118-8a74-11e8-80fa-7aa955ff9840.png)
![image](https://user-images.githubusercontent.com/27624208/42872333-927dd428-8a74-11e8-837b-aceaaa09cbe2.png)

#### Questions:
NA

Stylll reviewed (approved):
Great job. Keep it up.

ozimos committed:
feat(friends): add modals and endpoints

- Adds friends to the user collection
- Adds /api/friends endpoint to allow adding a user as friends and getting a user's friends
- Adds notifications to the navbar when a friend is added
- Adds a modal to search for users

[Finishes #158487973]
